ikarus/io/resultevaluators.hh

```diff
@@ -75,27 +104,121 @@ struct PrincipalStress
    */
-  static std::string name() { return "PrincipalStress"; }
+  constexpr static std::string name() { return "PrincipalStress"; }
 
   /**
    * \brief Get the number of components in the result
    * \return Number of components
    */
-  static int ncomps() { return dim; }
+  constexpr static int ncomps() { return dim; }
+};
+
+/**
+ * \brief Struct for calculating stress triaxiality
+ * \ingroup resultevaluators
+ * \details The Triaxiality struct provides a function call operator to calculate stress triaxiality.
+ * In 2D, this assumes a plane stress state
+ */
+struct Triaxiality
+{
+  /**
+   * \brief Calculate the result quantity (stress triaxiality)
+   * \param resultArray EigenVector containing the stress state in Voigt notation
+   * \param comp component of result (not used here)
+   * \tparam R Type of the matrix
+   * \return Triaxiality stress
+   */
+  template <typename R>
+  double operator()(const R& resultArray, [[maybe_unused]] const auto& pos, [[maybe_unused]] const auto& fe,
+                    [[maybe_unused]] const int comp) const {
+    auto sigeq = VonMises{}(resultArray, pos, fe, 0);
+    auto sigm  = HydrostaticStress{}(resultArray, pos, fe, 0);
+    return sigm / sigeq;
+  }
+  /**
+   * \brief Get the name of the result type
+   * \return String representing the name
+   */
+  constexpr static std::string name() { return "Triaxiality"; }
+
+  /**
+   * \brief Get the number of components in the result  (always 1 for stress triaxiality)
+   * \return Number of components
+   */
+  constexpr static int ncomps() { return 1; }
+};
+
+/**
+ * \brief Struct for calculating the 2d polar stress. The center of the coordinate system is chosen to be the center of
+ * the corresponding reference geometry.
+  \ingroup resultevaluators
+ */
+struct PolarStress
+{
+  PolarStress(const Dune::FieldVector<double, 2>& origin)
+      : origin_(origin) {}
+
+  /**
+   * \brief Calculate the result quantity (von Mises stress)
+   * \param resultArray EigenVector containing the stress state in Voigt notation
+   * \param comp component of result
+   * \tparam R Type of the matrix
+   * \return von Mises stress
+   */
+  template <typename R>
+  double operator()(const R& resultArray, const auto& pos, const auto& fe, const int comp) const {
+    static_assert(R::CompileTimeTraits::RowsAtCompileTime == 3, "PolarStress is only valid for 2D.");
+    if (comp > 2)
+      DUNE_THROW(Dune::RangeError, "PolarStress: Comp out of range.");
+
+    // Offset to center the coordinate system in the reference geometry
+    Dune::FieldVector<double, 2> posGlobal = fe.geometry().global(pos) - origin_;
+    auto theta                             = std::atan2(posGlobal[1], posGlobal[0]);
+
+    const auto sigma = fromVoigt(resultArray, false);
+    Eigen::Rotation2D<double> r(theta);
+
+    // deliberately not evaluating this that it stays an expression for below
+    auto polarStress = r.inverse() * sigma * r;
```

Comment:
Ah there is no transpose method for `Rotation2D`, sorry for this mistake before.